lecture_1/lecture_1_passed/passed.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sethgrid/pester"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+type passedAPIResponse struct{
+	Name string
+	Age int
+	Passed bool
+	Skills []string
+}
+
+
+
+const passedUrl = "https://run.mocky.io/v3/f7ceece5-47ee-4955-b974-438982267dc8"
+
+func linearBackoff(retry int) time.Duration {
+	return time.Duration(retry) * time.Second
+}
+
+func writingToFile(f *os.File, decodedContent []passedAPIResponse){
+	for i := range decodedContent{
+		if decodedContent[i].Passed{
+			var found = false
+			//var knownSkills []string
+			var knownSkillText = strings.Join(decodedContent[i].Skills, ", ")
+			for _,val := range decodedContent[i].Skills{
+				//knownSkills = append(knownSkills, val)
+
+				if val == "Java" || val == "Go"{
+					found = true
+				}
+			}
+			if found{
+				defer f.WriteString(fmt.Sprintf("%s - %v\n", decodedContent[i].Name, knownSkillText))
+			}
```

Comment:
This is a creative solution, I like how nicely it uses Go constructs. However, it's generally better to only open the file for writing once. If you had a lot of entries in the JSON, your program would run a lot slower due to all the file accessing.

In other words, you should first create the string and then save the entire thing to the file at once. For creating the string, take a look at [this](https://www.geeksforgeeks.org/strings-join-function-in-golang-with-examples/).

Reply:
Thank you for all inputs, i think i fixed all problems that were mentioned. Here is the link to an improved version: 
https://github.com/Rzucky/code-cadets-2021/pull/1/commits/bc9d70a17a549a0f43995c32cf9d0ea53e8d75e0
i will also post it on google docs since Slack isn't working